build.rs

```diff
@@ -1,39 +1,53 @@
-extern crate gcc;
+extern crate cc;
 
 use std::env;
 
 fn main() {
     let target = env::var("TARGET").unwrap();
-    let mut base_config = gcc::Build::new();
-    base_config.cpp(true)
+    let mut base_config = cc::Build::new();
+    base_config
+        .cpp(true)
         .include("bitcoin/src")
         .include("bitcoin/src/secp256k1/include")
-        .flag("-g").flag("-O2").flag("-std=c++11").flag("-Wno-unused-parameter")
+        .opt_level(2)
+        .debug(true)
+        .extra_warnings(true)
         .define("__STDC_FORMAT_MACROS", None);
+
+    let tool = base_config.get_compiler();
+    if tool.is_like_msvc() {
+        base_config.flag("/std:c++14");
```

Comment:
Shouldnt we use C++11 not 14?

Reply:
visual studio can specify the c++ standard version only from c++14.
https://msdn.microsoft.com/en-us/library/mt490614.aspx

Reply:
Hmm, ok, I've been informed -std=c++11 doesn't actually do anything in modern gccs anyway, so I guess it shouldnt matter. Better to leave 14, though.

Reply:
Default gcc from travis is too old and doesn't understands `-std=c++14`.